File: VB/WindowsApplication1/MyRibbonControl/CanMinimizePageGroupEventArgs.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.XtraBars.Ribbon.ViewInfo

Namespace WindowsApplication1
	Public Class CanMinimizePageGroupEventArgs
		Inherits EventArgs
		Private _canMinimize As Boolean
		Private _pageGroup As RibbonPageGroup
		Private _panelViewInfo As RibbonPanelViewInfo

		Public Sub New(ByVal canMinimize As Boolean, ByVal pageGroup As RibbonPageGroup, ByVal panelViewInfo As RibbonPanelViewInfo)
			MyBase.New()
			_canMinimize = canMinimize
			_pageGroup = pageGroup
			_panelViewInfo = panelViewInfo
		End Sub

		Public ReadOnly Property PageGroup() As RibbonPageGroup
			Get
				Return _pageGroup
			End Get
		End Property

		Public Property CanMinimize() As Boolean
			Get
				Return _canMinimize
			End Get
			Set(ByVal value As Boolean)
				_canMinimize = value
			End Set
		End Property
		Public ReadOnly Property PanelViewInfo() As RibbonPanelViewInfo
			Get
				Return _panelViewInfo
			End Get
		End Property

	End Class
End Namespace

File: CS/WindowsApplication1/MyRibbonControl/CanMinimizePageGroupEventArgs.cs
using System;
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraBars.Ribbon.ViewInfo;

namespace WindowsApplication1
{
    public class CanMinimizePageGroupEventArgs : EventArgs
    {
        private bool _canMinimize;
        private RibbonPageGroup _pageGroup;
        private RibbonPanelViewInfo _panelViewInfo;

        public CanMinimizePageGroupEventArgs(bool canMinimize, RibbonPageGroup pageGroup, RibbonPanelViewInfo panelViewInfo)
            : base()
        {
            _canMinimize = canMinimize;
            _pageGroup = pageGroup;
            _panelViewInfo = panelViewInfo;
        }

        public RibbonPageGroup PageGroup
        {
            get
            {
                return _pageGroup;
            }
        }

        public bool CanMinimize
        {
            get
            {
                return _canMinimize;
            }
            set
            {
                _canMinimize = value;
            }
        }
        public RibbonPanelViewInfo PanelViewInfo
        {
            get
            {
                return _panelViewInfo;
            }
        }

    }
}
